lib/src/commands/packages/commands/check/commands/licenses.dart

```diff
@@ -22,11 +22,27 @@ class PackagesCheckLicensesCommand extends Command<int> {
     PubLicense? pubLicense,
   })  : _logger = logger ?? Logger(),
         _pubLicense = pubLicense ?? PubLicense() {
-    argParser.addFlag(
-      'ignore-failures',
-      help: 'Ignore any license that failed to be retrieved.',
-      negatable: false,
-    );
+    argParser
+      ..addFlag(
+        'ignore-failures',
+        help: 'Ignore any license that failed to be retrieved.',
+        negatable: false,
+      )
+      ..addMultiOption(
+        'dependency-type',
+        help: 'The type of dependencies to check licenses for.',
+        allowed: [
+          'direct-main',
+          'direct-dev',
+          'transitive',
+        ],
```

Comment:
Do we want an `all`?

Reply:
Is a multi-option so for all you should do: `--dependency-type direct-main,direct-dev,transitive`. I think all might be adding unnecessary additional complexity? Might lead to additional questions from the user: "why can I use _all_ and _direct-main_"?